far/vmenu.cpp

```diff
@@ -140,25 +140,57 @@ void VMenu::ResetCursor()
 	GetCursorType(PrevCursorVisible,PrevCursorSize);
 }
 
+int find_nearest(std::ranges::contiguous_range auto const& Range, const int Pos, auto&& Pred, const bool GoBackward, const bool DoWrap)
+{
+	using namespace std::views;
+
+	assert(0 <= Pos && Pos < static_cast<int>(Range.size()));
+
+	const auto FindPos =
+		[&](const auto First, const auto Second)
+		{
+			const auto FindPosPart =
+				[&](const auto Part)
+				{
+					if (auto Filtered = Range | Part | filter(std::forward<decltype(Pred)>(Pred)))
```

Comment:
We have `FWD` macro that does exactly this.

Reply:
Will do (though macros are evil 🙂).

P.S. Please post something here when you finish commenting so that I know when I can push.

Reply:
>  I'm not sure that using forwarding is technically correct here, since FindPosPart is called _twice_.

Argh! You are probably right.